subcommand/server-acl-init/create_or_update.go

```diff
@@ -10,14 +10,40 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// createACL creates a policy with rules and name, creates an ACL token for that
-// policy and then writes the token to a Kubernetes secret.
-func (c *Command) createACL(name, rules string, consulClient *api.Client) error {
+// createLocalACL creates a policy and acl token for this dc (datacenter), i.e.
+// the policy is only valid for this datacenter and the token is a local token.
+func (c *Command) createLocalACL(name, rules, dc string, consulClient *api.Client) error {
+	return c.createACL(name, rules, true, dc, consulClient)
+}
+
+// createGlobalACL creates a global policy and acl token. The policy is valid
+// for all datacenters and the token is global. dc must be passed because the
+// policy may have the datacenter name appended.
+func (c *Command) createGlobalACL(name, rules, dc string, consulClient *api.Client) error {
+	return c.createACL(name, rules, false, dc, consulClient)
+}
+
+// createACL creates a policy with rules and name. If localToken is true then
+// the token will be a local token. If dc is not empty, the policy will
+// be scoped to just that datacenter, otherwise it will be global.
+// The token will be written to a Kubernetes secret.
+func (c *Command) createACL(name, rules string, localToken bool, dc string, consulClient *api.Client) error {
 	// Create policy with the given rules.
+	policyName := fmt.Sprintf("%s-token", name)
+	if c.flagEnableACLReplication {
+		// If performing ACL replication, we must ensure policy names are
+		// globally unique so we append the datacenter name.
+		policyName += fmt.Sprintf("-%s", dc)
+	}
+	var datacenters []string
+	if localToken {
```

Comment:
Should this be:
```suggestion
	if dc != "" {
```
?

Reply:
Currently we always pass the dc field, regardless of whether we're replicating or not. This is because we always retrieve it at the beginning of the bootstrap in to the `consulDC` variable and then always pass it through. I did this because I thought it best to have it always populated in case we want to use it in instances where we're not replicating. I could see someone later using that variable and expecting it to be populated.

Thus we can't rely on it being not-empty to indicate if the policy should be scoped to a single DC and instead need to rely on the localToken variable.

The docs on this function are wrong though.

We could change it to not populating `consulDC` when not replicating or always passing `""` when not replicating. Something like
```go
func (c *Command) createGlobalACL(name, rules, dc string, consulClient *api.Client) error {
	if !c.flagEnableACLReplication {
		dc = ""
	}
	return c.createACL(name, rules, false, dc, consulClient)
}
```
would work.

Reply:
Got it! Yeah, that makes sense to me. Thanks for updating the comment too. 

One suggestion I'd make though is making sure `dc` is not empty before appending it to the `datacenters` array. Otherwise, we could create a policy with `Datacenters` equal to `[]string{""}`. Not entirely sure how Consul would behave in this case, but probably best to avoid it.